lib/transproc/coercions.rb

```diff
@@ -153,5 +153,25 @@ def to_time(value)
     def to_datetime(value)
       DateTime.parse(value)
     end
+
+    # Coerce value into an array containing tuples only
+    #
+    # If the source is not an array, or doesn't contain a tuple, returns
+    # an array with one empty tuple
+    #
+    # @example
+    #   Transproc(:to_tuples)[:foo]                  # => [{}]
+    #   Transproc(:to_tuples)[[]]                    # => [{}]
+    #   Transproc(:to_tuples)[[{ foo: :FOO, :bar }]] # => [{ foo: :FOO }]
+    #
+    # @param [Object] value
+    #
+    # @return [Array<Hash>]
+    #
+    def to_tuples(value)
+      array = value.is_a?(Array) ? Array[*value] : [{}]
```

Comment:
Interesting question. I didn't cover the edge case when value is a tuple. My example will ignore it, while yours will cover it into array. I think we could make a decision, what should be a proper behaviour:
```ruby
source = [{ foo: { bar: :BAR }, baz: :BAZ }]
fn = t(:group, :foo, [:baz])
```
Should it return (as in my implementation):
```ruby
[{ foo: [{ baz: :BAZ }]]
```
Or:
```ruby
[{ foo: [{ bar: :BAR, baz: :BAZ  }] }]
```
I have agruments in favor of any of those possibilities

1) Skip tuple, because the intention of grouping is to sequentially group attributes. There is no step, when grouping would provide single tuple (not array of tuples)

2) Preserve touple - because it stores values. Tbh this is a weak argument, because we cannot preserve anything.

I'd prefer my first decision (and fixed it in spec). But what do you and @solnic think?